op-challenger/game/fault/types/game.go

```diff
@@ -41,6 +45,17 @@ type claimEntry struct {
 	ParentContractIndex int
 }
 
+func NewClaimEntry(claim Claim) claimEntry {
+	return claimEntry{
+		ClaimData:           claim.ClaimData,
+		ParentContractIndex: claim.ParentContractIndex,
+	}
+}
+
+func (c claimEntry) Hash() common.Hash {
+	return HashFromClaimData(c.Position.ToGIndex(), c.Value, c.ParentContractIndex)
+}
```

Comment:
I don't think the `claimEntry` struct is giving us any value here. Since we need to move to a derived identifier like a hash for claims, we should just remove the struct entirely.  We'd wind up with just:

```
func computeClaimID(claim Claim) common.Hash {
	return crypto.Keccak256Hash(
		claim.Position.ToGIndex().Bytes(),
		claim.Value.Bytes(),
		big.NewInt(int64(claim.ParentContractIndex)).Bytes(),
	)
}
```

and all the uses of `claimEntry` in the file become `common.Hash` (including for `children` in `extendedClaim`).

And for good measure I'd use a type alias like `type claimID common.Hash` so it's clear when we're using a claim ID and not just any hash.

Reply:
Thank you for doing this in your pr!